internal/datanode/channel_meta.go

```diff
@@ -219,6 +240,22 @@ func (c *ChannelMeta) listCompactedSegmentIDs() map[UniqueID][]UniqueID {
 	return compactedTo2From
 }
 
+func (c *ChannelMeta) listSegmentIDsToSync(ts Timestamp) []UniqueID {
+	c.segMu.RLock()
+	defer c.segMu.RUnlock()
+
+	segIDsToSync := make([]UniqueID, 0)
+	for segID, seg := range c.segments {
+		for _, policy := range c.syncPolicies {
+			if policy(seg, ts) {
+				segIDsToSync = append(segIDsToSync, segID)
+				seg.lastSyncTs = ts
```

Comment:
Other types of sync should change lastSyncTs too, and it should be changed `AFTER` segment syncing is complete.

Reply:
make sense